### PR TITLE
Port of optimum-setup - lifecycles to docs

### PR DESCRIPTION
**Pull request by harrisonmeister**
This work involves a port of the lifecycles section of the [Optimum Setup Book](https://github.com/OctopusDeploy/OptimumSetupBook/blob/master/manuscript/Lifecycles.md)

- Added default lifecycle information
- Adding Phase definitions
- Adding example lifecycles (could be a sub-section)

### CODE REVIEW

File: docs/releases/lifecycles/index.md [hunk 1]
```diff
@@ -57,3 +67,74 @@ If you want to be able to deploy to any environment at any time, then simply cre
...
+
+It typically consists of just one phase and one environment, also called Maintenance. You can include this environment in all deployment targets you want to run these tasks against. You can also split them up into the Development, Testing, Staging, and Production environments if you want to run the tasks for targets in those environments at different times.
+
+![Maintenance lifecycle](images/maintenance-lifecycle.png "width=500")
+
+## Recommendations {#lifecycle-recommendations}
+
+When configuring your lifecycles, here are some tips to consider:
+
+- Update the default lifecycle to define the phases you need. This makes configuring and maintaining your Octopus Server easier.
```
Comment (ryanrousseau) [inline docs/releases/lifecycles/index.md:137]: I need to turn my script for this into an api example

File: docs/releases/lifecycles/index.md [hunk 2]
```diff
@@ -57,3 +67,74 @@ If you want to be able to deploy to any environment at any time, then simply cre
...
+
+## Examples {#lifecycle-examples}
+
+In this section, we cover some lifecycle examples and their included phases.
+
+### Hotfix lifecycle
+
+A hotfix lifecycle is useful when you have a critical bug-fix that needs to be deployed quickly. In this scenario, lower environments such as Development and Testing are skipped. Deploying to both typically takes too much time.
+
+It's recommended to follow good deployment practices and validate any changes before pushing to production. To match this, a hotfix lifecycle usually has just two phases, Staging and Production. Software with the bug fix is validated in Staging, and then promited to Production. However your lifecycle may be different to reflect how you decide to handle hotfixes.
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md:121]: :smile: I added and removed the `However` a number of times. So I'm cool removing it.

File: docs/releases/lifecycles/index.md (hunk 1 above)
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md:137]: That'd be good, we could link to it in this section once its ready?

File: docs/releases/lifecycles/index.md [hunk 3]
```diff
@@ -57,3 +67,74 @@ If you want to be able to deploy to any environment at any time, then simply cre
...
+
+![Hotfix lifecycle](images/hotfix-lifecycle.png "width=500")
+
+### Maintenance lifecycle
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md:125]: See [this conversation](https://octopusdeploy.slack.com/archives/C01DS0NM9AT/p1605536275031400) (private link). I think we should add a reference to Runbooks in this section to highlight its possible to model this in an alternative way.

File: docs/releases/lifecycles/index.md [hunk 4]
```diff
@@ -57,3 +67,74 @@ If you want to be able to deploy to any environment at any time, then simply cre
...
+
+This allows you to explicitly configure the default lifecycle for deploying your software.
+
+## Examples {#lifecycle-examples}
+
+In this section, we cover some lifecycle examples and their included phases.
+
+### Hotfix lifecycle
+
+A hotfix lifecycle is useful when you have a critical bug-fix that needs to be deployed quickly. In this scenario, lower environments such as Development and Testing are skipped. Deploying to both typically takes too much time.
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md:119]: Agreed.

File: docs/releases/lifecycles/index.md [hunk 5]
```diff
@@ -12,6 +12,16 @@ Lifecycles enable a number of advanced deployment workflow features:
 - **Automate deployment to specific environments**: for example, automatically deploy to *test* as soon as a release is created.
 - **Retention policies**: specify the number of releases to keep depending on how far they have progressed through the lifecycle.
 
+:::hint
+Lifecycles don't apply to [Runbooks](/docs/runbooks/index.md). Learn more about the [differences between Runbooks and Deployments](/docs/runbooks/runbooks-vs-deployments/index.md).
+:::
+
+## Phases
+
+A phase represents a stage in your deployment lifecycle. You deploy to phases in order, and you must have a successful deployment in one phase to move to the next phase. For example, there must be a successful deployment to the Development phase before you can proceed to the Testing phase.
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md]: Agreed, re-worded in 70d08178caa46c3eaae5d15007fd9757ac8592ce

File: docs/releases/lifecycles/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/releases/lifecycles/index.md:126]: Added a `success` callout right under the header in 5c5c15fed16fcae279d01107c6c08a6658ebcb02

**Review by ryanrousseau — APPROVED** [review]

**Pull request merged**